### PR TITLE
[RuntimePermissions] Modernize watch app

### PR DESCRIPTION
**Pull request by alexvanyo**
Updates the wear portion of the `RuntimePermissions` sample, converting from Java to Kotlin and introducing coroutines, `ViewBinding`, and the `androidx` wear library.

### CODE REVIEW

File: RuntimePermissionsWear/Wearable/src/main/res/drawable/action_background.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (codingjeremy) [inline RuntimePermissionsWear/Wearable/src/main/res/drawable/action_background.xml:1]: Missing open source headers.

File: RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_send_to_mobile.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (codingjeremy) [inline RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_send_to_mobile.xml:1]: Missing open source headers.

File: RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_very_dissatisfied.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (codingjeremy) [inline RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_very_dissatisfied.xml:1]: Missing open source headers.

File: RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_very_satisfied.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (codingjeremy) [inline RuntimePermissionsWear/Wearable/src/main/res/drawable/ic_very_satisfied.xml:1]: Missing open source headers.

File: RuntimePermissionsWear/Wearable/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -1,5 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (codingjeremy) [inline RuntimePermissionsWear/Wearable/src/main/res/layout/activity_main.xml:1]: Nit, I know studio does this, but the comment should really start on the next line. ^_^;;

File: RuntimePermissionsWear/Wearable/src/main/res/values/colors.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="action_blue">#0086D4</color>
+    <color name="permission_granted">#20B000</color>
+    <color name="permission_denied">#B00020</color>
+</resources>
```
Comment (codingjeremy) [inline RuntimePermissionsWear/Wearable/src/main/res/values/colors.xml:6]: Empty new line needed at the end (or it shows up as a red negative in github.

File: RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/IncomingRequestWearService.kt [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    private val scope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+
+    override fun onDestroy() {
+        scope.cancel()
+        super.onDestroy()
+    }
+
+    override fun onMessageReceived(messageEvent: MessageEvent) {
+        Log.d(TAG, "onMessageReceived(): $messageEvent")
+        scope.launch {
```
[the remark this answers is not in the document]
Comment (alexvanyo) [reply RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/IncomingRequestWearService.kt:50]: I can add a comment here!

It is a bit confusing that `suspend` gets passed down through a bunch of functions to the place where it's actually necessary, but deferring the `scope.launch` makes it a bit harder to reason about the order of things.

Being able to initially switch to a suspending context implies "the work we need to do based on the message is asynchronous, so now everything we're doing might be asynchronous." I'll write something to that affect here in a comment!

File: RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/RequestPermissionOnPhoneActivity.kt [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityRequestPermissionOnPhoneBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.openOnPhoneContainer.setOnClickListener {
+            setResult(RESULT_OK)
+            finish()
+        }
```
[the remark this answers is not in the document]
Comment (alexvanyo) [reply RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/RequestPermissionOnPhoneActivity.kt:41]: Yeah, my preference is the other way so I think I'll leave it.
The unused-but-necessary `View` argument bothers me a bit.

**Comment by alexvanyo** [comment]
@codingjeremy should be ready for another look, I added/updated all open source headers, added new lines and added a comment around the coroutines usage in the `Service`.

File: RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/RequestPermissionOnPhoneActivity.kt (hunk 8 above)
[the remark this answers is not in the document]
Comment (codingjeremy) [reply RuntimePermissionsWear/Wearable/src/main/java/com/example/android/wearable/runtimepermissions/RequestPermissionOnPhoneActivity.kt:41]: Ah, I see, that makes sense. SGTM.

File: RuntimePermissionsWear/Wearable/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (codingjeremy) [reply RuntimePermissionsWear/Wearable/src/main/res/layout/activity_main.xml]: Wasn't fixed but I'll approve since maybe it was missed?

**Review by codingjeremy — APPROVED** [review]

**Comment by alexvanyo** [comment]
Thanks @codingjeremy !

**Pull request merged**